### PR TITLE
DRTII-960 removing global execution context and adding some logs

### PR DESCRIPTION
**Pull request by ngu04**

### CODE REVIEW

**Review by richbirch — COMMENTED** [review]
It would be good to know what the changes are trying to do. If you're attempting to fix something then could we have a test that fails without the changes and passes with the changes?

Is there a good reason to remove the global execution context? We're not using different configurations, so it's quite a lot less code to use the global

**Comment by ngu04** [comment]
> It would be good to know what the changes are trying to do. If you're attempting to fix something then could we have a test that fails without the changes and passes with the changes?

> Is there a good reason to remove the global execution context? We're not using different configurations, so it's quite a lot less code to use the global

Sorry , I should have explain . The changes are not fixing anything at moment  . Changes are like based on some of suspect  and good practise and trying to get more details with logs
>  It is good practise to avoid global execution context where ever possible https://www.scala-lang.org/api/2.13.3/scala/concurrent/ExecutionContext.html 

> Stream is deprecated so used LazyList 

> I have added logs at couple of places to see more details when application get stuck. Application may be getting stuck because the non 200 OK response is not logged and at same time isReady is still true .  

I am hoping to have more finding with the changes .

**Review by richbirch — CHANGES_REQUESTED** [review]
If you move the implicit executioncontexts to the class level then you don't need to add them to all the functions. That should tidy things up quite a lot - they won't be needed in the trait functions etc

File: src/main/scala/uk/gov/homeoffice/cirium/services/feed/Cirium.scala [hunk 1]
```diff
@@ -128,15 +143,15 @@ object Cirium {
 
     val latestItemActor: ActorRef = system.actorOf(Props(classOf[CiriumLastItemActor]), "latest-item-actor")
 
-    def start(step: Int): Future[Source[CiriumTrackableStatus, Cancellable]] = {
+    def start(step: Int)(implicit executionContext: ExecutionContext): Future[Source[CiriumTrackableStatus, Cancellable]] = {
```
Comment (richbirch) [inline src/main/scala/uk/gov/homeoffice/cirium/services/feed/Cirium.scala:146]: Could move these two to the class as well as we already have an implicit for the actor system

**Review by richbirch — APPROVED** [review]
:boom:

**Pull request merged**